include/AVI20/Read/IStream.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <AVI20/AVI20Defs.h>
+#include <AVI20/Namespace.h>
+
+#include <exception>
+
+#include <stdint.h>
+
+NAMESPACE_AVI20_BEGIN
+struct AVISTDINDEX;
+struct AVISTDINDEX_ENTRY;
+struct AVISUPERINDEX;
+struct AVISUPERINDEXENTRY;
+struct BITMAPINFOHEADER;
+struct MainHeader;
+struct MediaStreamHeader;
+struct WAVEFORMATEX;
+NAMESPACE_AVI20_END
+
+NAMESPACE_AVI20_READ_BEGIN
+
+class IStream
+{
+public:
+   virtual bool IsNULL() const = 0;
+
+   //template<class T> virtual T Read() = 0;
```

Comment:
I'm not crazy about having all these type-specific methods but there's no such thing as a templated pure-virtual method.